### ISSUE TITLE
[Queued] Verify #77 and #78 in next nightly sweep — confirm break counts are stable across runs and the layerdrop/mamba sub-classification regex still extracts cleanly

### ISSUE THREAD

**Issue by penguinwu** [penguinwu/oss-model-graph-break-corpus#79]
**Queued at:** 2026-04-29 10:45 UTC
**Status:** Backlog (queued — not started)

---

(No additional details. Add a comment when picking up so the context is captured before work starts.)

**Comment by penguinwu** [comment]
## 2026-05-03 nightly verification (per the queue)

Picked this up while writing the 2026-05-03 nightly report. Comparing the 2026-04-26 baseline (when #77/#78 were filed) against the 2026-05-03 nightly using both naive substring matching and the proper `tools/file_issues.py:classify_breaks` style.

### Wrapper-pattern issues (#102, #103) — naive count divergence

Naive substring count (`reason.count("Failed to handle graph break gracefully")` and `reason.count("we cannot resume from")`) on the 04-26 baseline does NOT reproduce the numbers in the issue bodies. Suggests the issues were counted via a different filter (e.g., one mode only, or grouped by break-event vs. break-occurrence):

| Issue | Claim (mod / brk) | My naive count 04-26 (mod / brk) | My naive count 05-03 (mod / brk) |
|---|---|---|---|
| #102 graceful_wrapper | 166 / 718 | 183 / 926 | 188 / 926 |
| #103 cannot_resume_wrapper | 107 / 345 | 132 / 497 | 136 / 499 |

Action item: re-derive via the same script that produced the original counts. The deep-dive doc at `experiments/2026-05-03-graph-break-deep-dive.md` claims 166 / 107 model counts; need to find that script (likely lives in `tools/` or was a one-off — search for the producer of those exact numbers).

### #77 layerdrop pattern

Naive regex `dropout_probability\s*<\s*self\.layerdrop` on the 04-26 baseline (both modes):

- 04-26: 71 models, 398 break occurrences (mode-mixed)
- 05-03: 74 models, 419 break occurrences (mode-mixed)

The issue claims **89 models / 466 train-mode breaks**. The discrepancy is mostly because:
1. The issue is train-mode-restricted; my count mixed both modes
2. My regex may miss subtly-formatted variants of the layerdrop expression

Net direction: **+3 models, +21 breaks** week-over-week (slight regression; layerdrop pattern still dominant).

### #78 mamba/linear-attn mask pattern

Naive regex `torch\.all\([^)]*attention_mask\s*==\s*1` returned **0 / 0** on both 04-26 and 05-03. The issue claims 20 models / 204 breaks. My regex does not match what's stored in `break_reasons[i].reason`.

Need to inspect a known mamba-family model's break_reasons to derive the correct pattern. The actual reason text format is structured like:

```
Graph break in user code at .../models/mamba/modeling_mamba.py:NNN
Graph Break Reason: Encountered graph break when attempting to trace ...
Explanation: Detected data-dependent branching ...
Hint: ...
```

The mask-all-ones check probably surfaces under a generic `Detected data-dependent branching` explanation, not as a literal `torch.all(attention_mask == 1)` substring. Sub-classification needs to inspect the surrounding source-line context (the `at filename:NNN` part) rather than the explanation string itself.

### Recommendation

This issue is correctly diagnosing a verification gap. To close it cleanly:

1. Find / restore the script that produced the 166 / 107 / 89 / 20 numbers (it lives in commit history or a one-off `/tmp` script that wasn't checked in)
2. Convert to a checked-in tool under `tools/verify_issue_counts.py` or extend `tools/file_issues.py`
3. Run on both nightlies, compare, post counts in this comment
4. For #78 specifically: use file-path classification (`if "modeling_mamba" in location`) instead of explanation-text matching

I have NOT made these tool changes — flagging for whoever picks this up next. The 2026-05-03 nightly report (`experiments/2026-05-03-nightly-report.md`) carries this same caveat.
